src/utils.js

```diff
@@ -213,6 +216,14 @@ export function isMultiSelect(schema) {
   return Array.isArray(schema.items.enum) && schema.uniqueItems;
 }
 
+export function isFixedItems(schema) {
+  return Array.isArray(schema.items) && schema.items.length > 0 && schema.items.every(item => isObject(item));
+}
+
+export function allowAdditionalItems(schema) {
+  return isObject(schema.additionalItems);
+}
```

Comment:
In the spec it says:
> 8.2.3.1.  If items is a schema, then the child instance must be valid against this schema, regardless of its index, and regardless of the value of "additionalItems".

Therefore it is only valid when `items` is an array, and when `additionalItems` is `true` it implies **any value** of additional items is valid. IMO falling back to `items` or `items[0]` are not best choices. (Maybe we need a input widget for "any" or "raw"?)